storage-api/src/java/org/apache/hadoop/hive/common/ValidReadTxnList.java

```diff
@@ -137,48 +151,94 @@ public String writeToString() {
     return buf.toString();
   }
 
+  /**
+   * txnlist is represented in ranges like this: 10-20,14,16-50
```

Comment:
Please add some more comment about the constant "4" and the meaning of that.
Otherwise I think this is good.

Thanks,
Peter